include/nbl/video/IPhysicalDevice.h

```diff
@@ -22,6 +22,294 @@
+                result.accelerationStructureVertex = accelerationStructureVertex ^ other.accelerationStructureVertex;
+                return result;
+            }
+
+            inline bool operator<(const SUsage& other) const
+            {
+                if (vertexAttribute && !other.vertexAttribute) return false;
+                if (bufferView && !other.bufferView) return false;
+                if (storageBufferView && !other.storageBufferView) return false;
+                if (storageBufferViewAtomic && !other.storageBufferViewAtomic) return false;
+                if (accelerationStructureVertex && !other.accelerationStructureVertex) return false;
+                return true;
+            }
+
+            inline bool operator == (const SUsage& other) const
+            {
+                return
+                    (vertexAttribute == other.vertexAttribute) &&
+                    (bufferView == other.bufferView) &&
+                    (storageBufferView == other.storageBufferView) &&
+                    (storageBufferViewAtomic == other.storageBufferViewAtomic) &&
+                    (accelerationStructureVertex == other.accelerationStructureVertex);
+            }
+        };
+        
+        inline SUsage& operator[](const asset::E_FORMAT idx)
+        {
+            return m_usages[idx];
+        }
+
+        inline const SUsage& operator[](const asset::E_FORMAT idx) const
+        {
+            return m_usages[idx];
+        }
+
+        inline bool isSubsetOf(const SFormatBufferUsages& other) const
+        {
+            for(uint32_t i = 0; i < asset::EF_COUNT; ++i)
+                if(!(m_usages[i] < other.m_usages[i]))
+                    return false;
+            return true;
+        }
+
+        SUsage m_usages[asset::EF_COUNT] = {};
+    };
+}
+
+
+namespace std {
+
+    template<>
+    struct ::std::hash<nbl::video::detail::SFormatImageUsages::SUsage>
+    {
+        inline uint32_t operator()(const nbl::video::detail::SFormatImageUsages::SUsage& i) const
+        {
+            return
+                i.sampledImage |
+                (i.storageImage << 1) |
+                (i.storageImageAtomic << 2) |
+                (i.attachment << 3) |
+                (i.attachmentBlend << 4) |
+                (i.blitSrc << 5) |
+                (i.blitDst << 6) |
+                (i.transferSrc << 7) |
+                (i.transferDst << 8) |
+                (i.log2MaxSamples << 9);
+        }
+    };
+
+    template<>
+    struct ::std::hash<nbl::video::detail::SFormatBufferUsages::SUsage>
+    {
+        inline uint32_t operator()(const nbl::video::detail::SFormatBufferUsages::SUsage& b) const
+        {
+            return
+                b.vertexAttribute |
+                (b.bufferView << 1) |
+                (b.storageBufferView << 2) |
+                (b.storageBufferViewAtomic << 3) |
+                (b.accelerationStructureVertex << 4);
+        }
+    };
```

Comment:
bit confusing that you can use std::hash before its declared when it was placed into a template :? main reason why I moved SFormatBufferUsages and SFormatImageUsages into a separate namespace.